### ISSUE TITLE
Configuration for Page model with existing `RoutablePageMixin`

### ISSUE THREAD

**Issue by garrettc** [donhauser/wagtail-pdf#12]
I have a Wagtail Page model that's already using `RoutablePageMixin` to optionally accept an ISO 2 letter country code. 

Cut-down example:

```python
class CountryReportPage(SeoMixin, PdfViewPageMixin, RoutablePageMixin, Page):
    ROUTE_CONFIG = [
        ("html", r"^$"),
        ("pdf", r"^pdf/$"),
    ]

    # Wagtail field definitions

    @re_path(r"^$")
    def index(self, request):
        # Handle URLs of the form /reports/?country_code=ca
        country_code = request.GET.get('country_code')

        if country_code and validate_country(country_code):
            return redirect(self.url + country_code + '/', permanent=True)
        return super().serve(request)

    @re_path(r"^(?P<country_code>\w{2})/$")
    def country_report(self, request, country_code=None):
        if validate_country(country_code) is None:
            return redirect(self.url, permanent=True)

        # Custom code for `/reports/<country_code>`

```

With this set-up, the base page HTML and PDF routes work perfectly (`/reports/` and `/reports/pdf/`), but with the country code, only the HTML version works (`/reports/ca/`), `/reports/ca/pdf` returns a 404. 

I'm guessing I need to manipulate the `"pdf"` route in `country_report()` somehow, but I think I'm missing something as the exact syntax is eluding me. 

Can you offer any pointers on how to handle this scenario?

**Comment by donhauser** [comment]
> With this set-up, the base page HTML and PDF routes work perfectly (`/reports/` and `/reports/pdf/`), but with the country code, only the HTML version works (`/reports/ca/`), `/reports/ca/pdf` returns a 404.
> 
> I'm guessing I need to manipulate the `"pdf"` route in `country_report()` somehow, but I think I'm missing something as the exact syntax is eluding me.

You are right, the most straight forward way is to edit the `"pdf"` route in  `ROUTE_CONFIG`.  If your page has the route `/reports`, the `"pdf"` path under `ROUTE_CONFIG` will become `/reports/pdf/`, which then resolves to the function `serve_pdf()` (likewise `html` goes to `serve_html()`).

So let's have a short insight look on how `ROUTE_CONFIG` works:
Essentially,  `ROUTE_CONFIG` is just a override-supporting workaround for `@re_path()` (previously `@route()`) from  `wagtail.contrib.routable_page.models`. It adds configurable routes to the mode specific serve functions `serve_pdf()` and `serve_html()` that can be changed on inheritance, which would not possible if the `PdfViewPageMixin` would simply declare `@re_path`. 

```py
class CountryReportPage(SeoMixin, PdfViewPageMixin, RoutablePageMixin, Page):
    ROUTE_CONFIG = [
        ("foo", pattern),
        ("bar", other_pattern, args), # you can set the args of @re_path
    ]

    # becomes

    @re_path(pattern, "foo") # the second argument is the name
    def serve_foo(self, request, **kwargs):
        ...

    @re_path(other_pattern, *args)
    def serve_bar(self, request, **kwargs):
        ...
```

The relevant code for this can be found in `mixins.MultipleViewPageMixin.__init_subclass__()` if you are curious.

Long story short, I think what you are looking for is something like (_untested_)
```py

class CountryReportPage(SeoMixin, PdfViewPageMixin, RoutablePageMixin, Page):
    ROUTE_CONFIG = [
        ("html", r"^(?:(?P<country_code>\w{2})/)?$"),
        ("pdf",  r"^(?:(?P<country_code>\w{2})/)?pdf/$"),
    ]
```
where `(?:...)` is a non-capturing regex group.

Hope this helps :+1:

**Comment by garrettc** [comment]
Aha! Yes, an optional non-capturing regex, of course. That worked perfectly. Thank you, and thank you for this package. Great stuff.
